src/core.js

```diff
@@ -28,30 +28,34 @@ export const jsons2arrays = (jsons, headers) => {
 
 export const elementOrEmpty = (element) => element || element === 0 ? element : '';
 
-export const joiner = ((data,separator = ',') =>
- data.map((row, index) => row.map((element) => "\"" + elementOrEmpty(element) + "\"").join(separator)).join(`\n`)
+export const joiner = ((data, separator = ',', enclosingCharacter = '"') =>
+  data.map((row, index) => row
+    .map((element) => elementOrEmpty(element))
+    .map(column => `${enclosingCharacter}${column}${enclosingCharacter}`)
+    .join(separator))
+    .join(`\n`)
 );
 
-export const arrays2csv = ((data, headers, separator) =>
- joiner(headers ? [headers, ...data] : data, separator)
+export const arrays2csv = ((data, headers, separator, enclosingCharacter) =>
+ joiner(headers ? [headers, ...data] : data, separator, enclosingCharacter)
 );
 
-export const jsons2csv = ((data, headers, separator) =>
- joiner(jsons2arrays(data, headers), separator)
+export const jsons2csv = ((data, headers, separator, enclosingCharacter) =>
+ joiner(jsons2arrays(data, headers), separator, enclosingCharacter)
 );
 
-export const string2csv = ((data, headers, separator) =>
+export const string2csv = ((data, headers, separator, enclosingCharacter) =>
   (headers) ? `${headers.join(separator)}\n${data}`: data
 );
 
-export const toCSV = (data, headers, separator) => {
- if (isJsons(data)) return jsons2csv(data, headers, separator);
- if (isArrays(data)) return arrays2csv(data, headers, separator);
+export const toCSV = (data, headers, separator, enclosingCharacter) => {
+ if (isJsons(data)) return jsons2csv(data, headers, separator, enclosingCharacter);
+ if (isArrays(data)) return arrays2csv(data, headers, separator, enclosingCharacter);
  if (typeof data ==='string') return string2csv(data, headers, separator);
  throw new TypeError(`Data should be a "String", "Array of arrays" OR "Array of objects" `);
 };
 
-export const buildURI = ((data, uFEFF, headers, separator) => encodeURI(
-  `data:text/csv;charset=utf-8,${uFEFF ? '\uFEFF' : ''}${toCSV(data, headers, separator)}`
+export const buildURI = ((data, uFEFF, headers, separator, enclosingCharacter) => encodeURI(
```

Comment:
This function is used by two components : CSVLink (Link) and CSVDownload(Download). I think "src/components/Download.js" needs maintenance because of this update.

Reply:
I changed that component so that it passes the prop through to `buildURI`. Was there anything else that you think is missing?